### PR TITLE
Handle cyclical includes in lark parser

### LINKED ISSUE #5016: Missing check for circular imports in lark parser
**Describe the bug**
Lark parser does not consider circular includes. Should consider it in `handle_includes`

### PR DESCRIPTION
**Pull request by yngve-sk**
**Issue**
Resolves #5016

**Approach**
Compares current file with list of previously included files

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/enkf/lark_parser.py [hunk 1]
```diff
@@ -450,22 +455,32 @@ def handle_includes(tree: Tree, config_file: str):
...
-                val = os.path.normpath(os.path.join(config_dir, val))
+            if not os.path.isabs(file_to_include):
+                file_to_include = os.path.normpath(
+                    os.path.join(config_dir, file_to_include)
+                )
 
-            sub_tree = _parse_file(val, "INCLUDE")
-            handle_includes(sub_tree, val)
+            sub_tree = _parse_file(file_to_include, "INCLUDE")
+            if file_to_include in already_included_files:
```
Comment (eivindjahren) [inline src/ert/_c_wrappers/enkf/lark_parser.py:476]: Doesn't it make sense to parse the file only if it isn't a cyclical import?

Comment (yngve-sk) [reply src/ert/_c_wrappers/enkf/lark_parser.py:475]: Oops! Fixed now

**Comment by yngve-sk** [comment]
jenkins test this tease

**Comment by eivindjahren** [comment]
jenkins test er please

**Review by eivindjahren — APPROVED** [review]
:flags:

**Pull request merged**